File: src/main/java/fun/vs/fw/demo/repository/CategoryBudget.java
package fun.vs.fw.demo.repository;


import java.math.BigDecimal;

/**
 * Represents a budget summary for a specific expense category.
 * This immutable record is used to encapsulate the category name and the total amount
 * of financial transactions associated with that category.
 * <p>
 * It is primarily utilized as a projection in repository queries to aggregate
 * the total transaction amounts per category for a specific client.
 */
public record CategoryBudget(String category, BigDecimal totalAmount) {}

File: fvf4k/src/main/kotlin/fvf4k/demo/domain/model/CategoryBudget.kt
package fvf4k.demo.domain.model

/**
 * Represents a budget summary for a specific expense category.
 * This immutable record is used to encapsulate the category name and the total amount
 * of financial transactions associated with that category.
 *
 *
 * It is primarily used as a projection in repository queries to aggregate
 * the total transaction amounts per category for a specific client.
 */
data class CategoryBudget(val category: ExpenseCategory, val totalMoney: Money)